### ISSUE TITLE
Quarkus startup hangs in OidcRecorder

### ISSUE THREAD

**Issue by jendib** [quarkusio/quarkus#33993]
### Describe the bug

In some context (I'm still figuring out when) the server startup hangs indefinitely in OidcRecorder.java L143. It's only happening in production unfortunately. I managed to get this jstack:
```
"main" #1 prio=5 os_prio=0 cpu=2263.87ms elapsed=692.56s tid=0x00007f5c40024ec0 nid=0x3f waiting on condition  [0x00007f5c472d1000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.7/Native Method)
        - parking to wait for  <0x00000000f14a1d18> (a java.util.concurrent.CountDownLatch$Sync)
        at java.util.concurrent.locks.LockSupport.park(java.base@17.0.7/LockSupport.java:211)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.7/AbstractQueuedSynchronizer.java:715)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@17.0.7/AbstractQueuedSynchronizer.java:1047)
        at java.util.concurrent.CountDownLatch.await(java.base@17.0.7/CountDownLatch.java:230)
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:67)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.smallrye.mutiny.groups.UniAwait.indefinitely(UniAwait.java:46)
        at io.quarkus.oidc.runtime.OidcRecorder.createStaticTenantContext(OidcRecorder.java:143)
        at io.quarkus.oidc.runtime.OidcRecorder.setup(OidcRecorder.java:64)
        at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@17.0.7/Native Method)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@17.0.7/NativeMethodAccessorImpl.java:77)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@17.0.7/DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(java.base@17.0.7/Method.java:568)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
```

I think there should be a log at least, possibly a timeout or even this should be deferred to later to avoid a network call on startup.

### Output of `java -version`

17

### Quarkus version or git rev

3.1.0

**Comment by sberyozkin** [comment]
Hi, can you please try a connection-timeout property, sorry,  typing on my phone,  don't recall the full property name right now

**Comment by jendib** [comment]
@sberyozkin I tried `quarkus.oidc.connection-timeout=2S` but it's still frozen on startup. I tried to open a shell in the production container and `curl -vvv https://mykeycloak.example.com/realms/myrealm/.well-known/openid-configuration` and it answers instantly so it doesn't seem to be a network issue (there is not proxy or anything like that).

**Comment by sberyozkin** [comment]
Interesting, why does it hang then...

**Comment by sberyozkin** [comment]
@jendib Thanks for the investigation. There is likely a race condition somewhere masked by the added logging. I'll probably need to replace the last remaining indefinite wait

**Comment by paomian** [comment]
got same issue

```"main" #1 prio=5 os_prio=0 cpu=3601.16ms elapsed=441.89s tid=0x00007f5180025e10 nid=0x3f waiting on condition  [0x00007f51881bd000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.7/Native Method)
        - parking to wait for  <0x00000007b6b3b1e0> (a java.util.concurrent.CountDownLatch$Sync)
        at java.util.concurrent.locks.LockSupport.park(java.base@17.0.7/LockSupport.java:211)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.7/AbstractQueuedSynchronizer.java:715)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@17.0.7/AbstractQueuedSynchronizer.java:1047)
        at java.util.concurrent.CountDownLatch.await(java.base@17.0.7/CountDownLatch.java:230)
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:67)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.smallrye.mutiny.groups.UniAwait.indefinitely(UniAwait.java:46)
        at io.quarkus.oidc.runtime.OidcRecorder.createStaticTenantContext(OidcRecorder.java:143)
        at io.quarkus.oidc.runtime.OidcRecorder.setup(OidcRecorder.java:71)
        at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at com.greptime.Application.main(Application.java:10)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@17.0.7/Native Method)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@17.0.7/NativeMethodAccessorImpl.java:77)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@17.0.7/DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(java.base@17.0.7/Method.java:568)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
```

**Comment by paomian** [comment]
```
Jun 28, 2023 11:27:52 AM org.hibernate.resource.beans.container.spi.AbstractCdiBeanContainer stop
INFO: HHH10005004: Stopping BeanContainer : %s
Jun 28, 2023 11:27:52 AM io.quarkus.runtime.ApplicationLifecycleManager run
ERROR: Failed to start application (with profile [prod])
io.smallrye.mutiny.TimeoutException
	at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:64)
	at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
	at io.quarkus.oidc.runtime.OidcRecorder.createStaticTenantContext(OidcRecorder.java:143)
	at io.quarkus.oidc.runtime.OidcRecorder.setup(OidcRecorder.java:71)
	at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.OidcBuildStep$setup635434700.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at com.greptime.Application.main(Application.java:10)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
```
got same error in 3.1.2

**Comment by paomian** [comment]
when this commit https://github.com/quarkusio/quarkus/commit/508994ca6bbea7ae706a8431bfe42da21828f408 release?
